### ISSUE TITLE
Tune The PassGenerator

### ISSUE THREAD

**Issue by garethellis0** [UBC-Thunderbots/Software#388]
### Description of the task
Tune the various hyperparameters that the `PassGenerator` uses to converge to good passes.

### Outline the task in more detail
Look into tuning the following in the `PassGenerator`:
- param weights
- hyperparameters for the gradient descent optimizer that we use (see the class constructor for the gradient descent class)
- all the dynamic parameters used by the `PassGenerator`

The goal is to maximize convergence to good passes in game scenarios, so this will likely involve some work setting up situations in grsim to test.

### Acceptance criteria

- [ ] Aforementioned hyperparameters are tuned as best as possible

**Comment by MathewMacDougall** [comment]
@garethellis0 I think we can consider this complete?

**Comment by garethellis0** [comment]
Yup, closing.
